bfvmm/src/serial/src/Makefile

```diff
@@ -0,0 +1,90 @@
+#
+# Bareflank Hypervisor
+#
+# Copyright (C) 2015 Assured Information Security, Inc.
+# Author: Rian Quinn        <quinnr@ainfosec.com>
+# Author: Brendan Kerrigan  <kerriganb@ainfosec.com>
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+
+################################################################################
+# Cross Flags
+################################################################################
+
+CROSS_CC=~/opt/cross/bin/x86_64-elf-gcc
+CROSS_CXX=~/opt/cross/bin/x86_64-elf-g++
+CROSS_ASM=~/opt/cross/bin/nasm
+CROSS_LD=~/opt/cross/bin/x86_64-elf-ld
+
+CROSS_CCFLAGS=
+CROSS_CXXFLAGS=-c
+CROSS_LDFLAGS=-fPIC
+CROSS_ASMFLAGS=-f elf64
+
+CROSS_OBJDIR=.build/cross
+CROSS_OUTDIR=../../../bin/cross
+
+################################################################################
+# Native Flags
+################################################################################
+
+CC=gcc
+CXX=g++
+LD=g++
+ASM=nasm
+
+CCFLAGS=
+CXXFLAGS=-c
+LDFLAGS=-fPIC
+ASMFLAGS=-f elf64
+
+OBJDIR=.build/native
+OUTDIR=../bin/native
+
+################################################################################
+# Common
+################################################################################
+
+DEFINES=
+
+RM=rm -rf
+MD=mkdir -p
+
+################################################################################
+# Sources
+################################################################################
+
+TARGET_NAME=bf_serial
+TARGET_TYPE=lib
+TARGET_COMPILER=both
+
+SOURCES=serial_port_x86.cpp bf_portio.asm portio_x64.cpp
```

Comment:
Interesting on that note.... one thing I learned is that with a makefile VAR=$VAR2 doesn't just set VAR equal to VAR2, it makes them the same thing. If VAR2 changes, so does VAR. If you use +=, that doesn't happen